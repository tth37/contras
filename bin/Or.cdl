import Not from "Not.cdl"

@Head
NAME Or
INPUT a, b
OUTPUT out
SYMBOL NAND Nand1
SYMBOL Not Not1, Not2

@Body
Not1(a = a) => (out = na)
Not2(a = b) => (out = nb)
Nand1(a = na, b = nb) => (out = res)

@End
OUTPUT out = res

# .\contras.exe .\Or.cdl .\test\Or.in .\test\Or.out