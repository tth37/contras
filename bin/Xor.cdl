import And from "And.cdl"
import Or from "Or.cdl"
import Not from "Not.cdl"

@Head
NAME Xor
INPUT a, b
OUTPUT out
SYMBOL And And1, And2
SYMBOL Or Or1
SYMBOL Not Not1, Not2

@Body
Not1(a = a) => (out = na)
Not2(a = b) => (out = nb)
And1(a = a, b = nb) => (out = anb)
And2(a = na, b = b) => (out = bna)
Or1(a = anb, b = bna) => (out = res)

@End
OUTPUT out = res

# .\contras.exe .\Xor.cdl .\test\Xor.in .\test\Xor.out