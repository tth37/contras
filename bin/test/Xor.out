@Case
  a[0] = 0101
  b[0] = 0011
out[0] = 0110
